extensions/integration-tests/setEnvironmentVariables.js

```diff
@@ -81,74 +80,57 @@ if (!fs.existsSync(NOTEBOOK_PYTHON_INSTALL_PATH)) {
-const ENVAR_AZURE_SERVER_USERNAME = 'AZURE_SQL_USERNAME';
-const ENVAR_AZURE_SERVER_PASSWORD = 'AZURE_SQL_PWD';
-const ENVAR_STANDALONE_SERVER = 'STANDALONE_SQL';
-const ENVAR_STANDALONE_SERVER_USERNAME = 'STANDALONE_SQL_USERNAME';
-const ENVAR_STANDALONE_SERVER_PASSWORD = 'STANDALONE_SQL_PWD';
-const ENVAR_STANDALONE_SERVER_2019 = 'STANDALONE_SQL_2019';
-const ENVAR_STANDALONE_SERVER_USERNAME_2019 = 'STANDALONE_SQL_USERNAME_2019';
-const ENVAR_STANDALONE_SERVER_PASSWORD_2019 = 'STANDALONE_SQL_PWD_2019';
 const ENVAR_PYTHON_INSTALL_PATH = 'PYTHON_TEST_PATH';
 const ENVAR_RUN_PYTHON3_TEST = 'RUN_PYTHON3_TEST';
-
-// Mapping between AKV secret and the environment variable names
-const SecretEnVarMapping = [];
-SecretEnVarMapping.push([SECRET_AZURE_SERVER, ENVAR_AZURE_SERVER]);
-SecretEnVarMapping.push([SECRET_AZURE_SERVER_PASSWORD, ENVAR_AZURE_SERVER_PASSWORD]);
-SecretEnVarMapping.push([SECRET_AZURE_SERVER_USERNAME, ENVAR_AZURE_SERVER_USERNAME]);
-SecretEnVarMapping.push([SECRET_STANDALONE_SERVER, ENVAR_STANDALONE_SERVER]);
-SecretEnVarMapping.push([SECRET_STANDALONE_SERVER_PASSWORD, ENVAR_STANDALONE_SERVER_PASSWORD]);
-SecretEnVarMapping.push([SECRET_STANDALONE_SERVER_USERNAME, ENVAR_STANDALONE_SERVER_USERNAME]);
-SecretEnVarMapping.push([SECRET_STANDALONE_SERVER_2019, ENVAR_STANDALONE_SERVER_2019]);
-SecretEnVarMapping.push([SECRET_STANDALONE_SERVER_USERNAME_2019, ENVAR_STANDALONE_SERVER_USERNAME_2019]);
-SecretEnVarMapping.push([SECRET_STANDALONE_SERVER_PASSWORD_2019, ENVAR_STANDALONE_SERVER_PASSWORD_2019]);
+const ENVAR_SQL_2017_PASS = 'SQL_2017_PASS';
+const ENVAR_SQL_2019_PASS = 'SQL_2019_PASS';
+const ENVAR_AZURE_SQL_PASS = 'AZURE_SQL_PASS';
 
 // Set the values that are not stored in AKV here
 process.env[ENVAR_PYTHON_INSTALL_PATH] = NOTEBOOK_PYTHON_INSTALL_PATH;
 process.env[ENVAR_RUN_PYTHON3_TEST] = '1';
-
-const credential = new DefaultAzureCredential();
-const client = new SecretClient(AKV_URL, credential);
-
-async function main() {
-	for (const entry of SecretEnVarMapping) {
-		const secretName = entry[0];
-		const environmentVariable = entry[1];
-		try {
-			const result = await client.getSecret(secretName);
-			console.log(`${secretName}: ${result.value}`);
-			process.env[environmentVariable] = result.value;
-		} catch (err) {
-			console.error('An error occured while retrieving the value for secret:' + secretName);
-			console.error(err);
-		}
+process.env[ENVAR_SQL_2017_PASS] = generatePassword();
+process.env[ENVAR_SQL_2019_PASS] = generatePassword();
+process.env[ENVAR_AZURE_SQL_PASS] = generatePassword();
+
+// run the docker.ps1 powershell script, and wait for it to finish
+// show all output from the command
+const command = `powershell.exe -Command "Start-Process powershell.exe -ArgumentList '-Command', '$sql2017pass=''${process.env[ENVAR_SQL_2017_PASS]}'';$sql2019pass=''${process.env[ENVAR_SQL_2019_PASS]}'';$azuresqlpass=''${process.env[ENVAR_AZURE_SQL_PASS]}'';\\"${__dirname}\\dockerInstall.ps1\\"' -Verb RunAs"`;
```

Comment:
Looks like you're having one powershell process start a _second_ powershell process and splicing together some powershell snippets?  This should be a cleaner way to do the same thing:

```js
const command = `powershell -File "./dockerSetup.ps1" -Sql2017Password "${process.env[ENVAR_SQL_2017_PASS]}" -Sql2019Password "${process.env[ENVAR_SQL_2019_PASS]}" -SqlAzurePassword "${process.env[ENVAR_AZURE_SQL_PASS]}"`;
```

then in something like dockerHelper.ps1:

```ps
# has to be the first line (except for comments):
param($Sql2017Password, $Sql2019Password, $SqlAzurePassword)

function InstallDocker() {
    Write-Host "Installing Docker..."
    # replace with dockerInstall.ps1 contents
}

function RunDocker($Sql2017Password, $Sql2019Password, $SqlAzurePassword) {
    Write-Host "Launching 2017: $Sql2017Password"
    Write-Host "Launching 2019: $Sql2019Password"
    Write-Host "Launching Azure: $SqlAzurePassword"
    # replace with dockerContainers.ps1 contents
}

InstallDocker
RunDocker -Sql2017Password $Sql2017Password -Sql2019Password $Sql2019Password -SqlAzurePassword $SqlAzurePassword
```

Reply:
I did it this way because it needs admin privileges to run, and I couldn't find a way to run it like that without the Start-Process command